{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "import src\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene stable ID version</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Transcript stable ID version</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Transcript start (bp)</th>\n",
       "      <th>Transcription start site (TSS)</th>\n",
       "      <th>Transcript end (bp)</th>\n",
       "      <th>Transcript length (including UTRs and CDS)</th>\n",
       "      <th>Transcript support level (TSL)</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Transcript count</th>\n",
       "      <th>Transcript type</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000187961</td>\n",
       "      <td>ENSG00000187961.14</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>ENST00000338591.8</td>\n",
       "      <td>960584</td>\n",
       "      <td>965719</td>\n",
       "      <td>1</td>\n",
       "      <td>960584</td>\n",
       "      <td>960584</td>\n",
       "      <td>965719</td>\n",
       "      <td>2567</td>\n",
       "      <td>tsl1 (assigned to previous version 7)</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000187583</td>\n",
       "      <td>ENSG00000187583.10</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>ENST00000379410.7</td>\n",
       "      <td>966497</td>\n",
       "      <td>975865</td>\n",
       "      <td>1</td>\n",
       "      <td>966497</td>\n",
       "      <td>966497</td>\n",
       "      <td>975108</td>\n",
       "      <td>2404</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000187583</td>\n",
       "      <td>ENSG00000187583.10</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>ENST00000379407.7</td>\n",
       "      <td>966497</td>\n",
       "      <td>975865</td>\n",
       "      <td>1</td>\n",
       "      <td>966502</td>\n",
       "      <td>966502</td>\n",
       "      <td>975008</td>\n",
       "      <td>2194</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENSG00000188157.15</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>ENST00000379370.7</td>\n",
       "      <td>1020120</td>\n",
       "      <td>1056118</td>\n",
       "      <td>1</td>\n",
       "      <td>1020120</td>\n",
       "      <td>1020120</td>\n",
       "      <td>1056116</td>\n",
       "      <td>7326</td>\n",
       "      <td>tsl1 (assigned to previous version 6)</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>10</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000162571</td>\n",
       "      <td>ENSG00000162571.13</td>\n",
       "      <td>ENST00000379290</td>\n",
       "      <td>ENST00000379290.5</td>\n",
       "      <td>1173884</td>\n",
       "      <td>1197935</td>\n",
       "      <td>1</td>\n",
       "      <td>1173903</td>\n",
       "      <td>1173903</td>\n",
       "      <td>1197935</td>\n",
       "      <td>2283</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>TTLL10</td>\n",
       "      <td>7</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene stable ID Gene stable ID version Transcript stable ID  \\\n",
       "0  ENSG00000187961     ENSG00000187961.14      ENST00000338591   \n",
       "1  ENSG00000187583     ENSG00000187583.10      ENST00000379410   \n",
       "2  ENSG00000187583     ENSG00000187583.10      ENST00000379407   \n",
       "3  ENSG00000188157     ENSG00000188157.15      ENST00000379370   \n",
       "4  ENSG00000162571     ENSG00000162571.13      ENST00000379290   \n",
       "\n",
       "  Transcript stable ID version  Gene start (bp)  Gene end (bp)  Strand  \\\n",
       "0            ENST00000338591.8           960584         965719       1   \n",
       "1            ENST00000379410.7           966497         975865       1   \n",
       "2            ENST00000379407.7           966497         975865       1   \n",
       "3            ENST00000379370.7          1020120        1056118       1   \n",
       "4            ENST00000379290.5          1173884        1197935       1   \n",
       "\n",
       "   Transcript start (bp)  Transcription start site (TSS)  Transcript end (bp)  \\\n",
       "0                 960584                          960584               965719   \n",
       "1                 966497                          966497               975108   \n",
       "2                 966502                          966502               975008   \n",
       "3                1020120                         1020120              1056116   \n",
       "4                1173903                         1173903              1197935   \n",
       "\n",
       "   Transcript length (including UTRs and CDS)  \\\n",
       "0                                        2567   \n",
       "1                                        2404   \n",
       "2                                        2194   \n",
       "3                                        7326   \n",
       "4                                        2283   \n",
       "\n",
       "          Transcript support level (TSL) Gene name  Transcript count  \\\n",
       "0  tsl1 (assigned to previous version 7)    KLHL17                 5   \n",
       "1                                   tsl1   PLEKHN1                 5   \n",
       "2                                   tsl1   PLEKHN1                 5   \n",
       "3  tsl1 (assigned to previous version 6)      AGRN                10   \n",
       "4                                   tsl1    TTLL10                 7   \n",
       "\n",
       "  Transcript type Chromosome/scaffold name  \n",
       "0  protein_coding                        1  \n",
       "1  protein_coding                        1  \n",
       "2  protein_coding                        1  \n",
       "3  protein_coding                        1  \n",
       "4  protein_coding                        1  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = pd.concat((pd.read_excel(src.external_data_path / '20190723_hg38TSSprotcodingnoYnoMT_4Luca.xlsx', sheet_name=\"+strand\"),\n",
    "                   pd.read_excel(src.external_data_path / '20190723_hg38TSSprotcodingnoYnoMT_4Luca.xlsx', sheet_name=\"-strand\")),\n",
    "                 axis=0)\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene stable ID version</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Transcript stable ID version</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Transcript start (bp)</th>\n",
       "      <th>Transcript end (bp)</th>\n",
       "      <th>Transcript length (including UTRs and CDS)</th>\n",
       "      <th>Transcript support level (TSL)</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Transcript count</th>\n",
       "      <th>Transcript type</th>\n",
       "      <th>uid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959256</td>\n",
       "      <td>959257</td>\n",
       "      <td>-1</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>ENSG00000188976.11</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>ENST00000327044.7</td>\n",
       "      <td>944203</td>\n",
       "      <td>959309</td>\n",
       "      <td>944203</td>\n",
       "      <td>959256</td>\n",
       "      <td>2757</td>\n",
       "      <td>tsl1 (assigned to previous version 6)</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960584</td>\n",
       "      <td>960585</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000187961</td>\n",
       "      <td>ENSG00000187961.14</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>ENST00000338591.8</td>\n",
       "      <td>960584</td>\n",
       "      <td>965719</td>\n",
       "      <td>960584</td>\n",
       "      <td>965719</td>\n",
       "      <td>2567</td>\n",
       "      <td>tsl1 (assigned to previous version 7)</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966497</td>\n",
       "      <td>966498</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000187583</td>\n",
       "      <td>ENSG00000187583.10</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>ENST00000379410.7</td>\n",
       "      <td>966497</td>\n",
       "      <td>975865</td>\n",
       "      <td>966497</td>\n",
       "      <td>975108</td>\n",
       "      <td>2404</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966502</td>\n",
       "      <td>966503</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000187583</td>\n",
       "      <td>ENSG00000187583.10</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>ENST00000379407.7</td>\n",
       "      <td>966497</td>\n",
       "      <td>975865</td>\n",
       "      <td>966502</td>\n",
       "      <td>975008</td>\n",
       "      <td>2194</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000111</td>\n",
       "      <td>1000112</td>\n",
       "      <td>-1</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>ENSG00000188290.10</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>ENST00000304952.10</td>\n",
       "      <td>998962</td>\n",
       "      <td>1000172</td>\n",
       "      <td>998964</td>\n",
       "      <td>1000111</td>\n",
       "      <td>899</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>HES4</td>\n",
       "      <td>4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  tss_start  tss_end  strand   Gene stable ID Gene stable ID version  \\\n",
       "0  chr1     959256   959257      -1  ENSG00000188976     ENSG00000188976.11   \n",
       "0  chr1     960584   960585       1  ENSG00000187961     ENSG00000187961.14   \n",
       "1  chr1     966497   966498       1  ENSG00000187583     ENSG00000187583.10   \n",
       "2  chr1     966502   966503       1  ENSG00000187583     ENSG00000187583.10   \n",
       "1  chr1    1000111  1000112      -1  ENSG00000188290     ENSG00000188290.10   \n",
       "\n",
       "  Transcript stable ID Transcript stable ID version  Gene start (bp)  \\\n",
       "0      ENST00000327044            ENST00000327044.7           944203   \n",
       "0      ENST00000338591            ENST00000338591.8           960584   \n",
       "1      ENST00000379410            ENST00000379410.7           966497   \n",
       "2      ENST00000379407            ENST00000379407.7           966497   \n",
       "1      ENST00000304952           ENST00000304952.10           998962   \n",
       "\n",
       "   Gene end (bp)  Transcript start (bp)  Transcript end (bp)  \\\n",
       "0         959309                 944203               959256   \n",
       "0         965719                 960584               965719   \n",
       "1         975865                 966497               975108   \n",
       "2         975865                 966502               975008   \n",
       "1        1000172                 998964              1000111   \n",
       "\n",
       "   Transcript length (including UTRs and CDS)  \\\n",
       "0                                        2757   \n",
       "0                                        2567   \n",
       "1                                        2404   \n",
       "2                                        2194   \n",
       "1                                         899   \n",
       "\n",
       "          Transcript support level (TSL) Gene name  Transcript count  \\\n",
       "0  tsl1 (assigned to previous version 6)     NOC2L                 6   \n",
       "0  tsl1 (assigned to previous version 7)    KLHL17                 5   \n",
       "1                                   tsl1   PLEKHN1                 5   \n",
       "2                                   tsl1   PLEKHN1                 5   \n",
       "1                                   tsl1      HES4                 4   \n",
       "\n",
       "  Transcript type  uid  \n",
       "0  protein_coding    0  \n",
       "0  protein_coding    1  \n",
       "1  protein_coding    2  \n",
       "2  protein_coding    3  \n",
       "1  protein_coding    4  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "order = ['Chromosome/scaffold name', 'Transcription start site (TSS)', 'Transcription start site (TSS)', 'Strand']\n",
    "order = order + [c for c in genes.columns if c not in order]\n",
    "genes = genes[order]\n",
    "genes['Chromosome/scaffold name'] = 'chr' + genes['Chromosome/scaffold name'].astype(str)\n",
    "genes.columns = ['chr', 'tss_start', 'tss_end', 'strand'] + genes.columns[4:].tolist()\n",
    "genes = genes.sort_values(['chr', 'tss_start', 'tss_end'])\n",
    "genes['uid'] = np.arange(genes.shape[0], dtype=int)\n",
    "genes['tss_end'] += 1\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes[['chr', 'tss_start', 'tss_end', 'uid']].to_csv(src.interim_data_path / \"hg38_ensembl_gene_tss.bed\", sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>uid</th>\n",
       "      <th>strand</th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene stable ID version</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Transcript stable ID version</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Transcript start (bp)</th>\n",
       "      <th>Transcript end (bp)</th>\n",
       "      <th>Transcript length (including UTRs and CDS)</th>\n",
       "      <th>Transcript support level (TSL)</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Transcript count</th>\n",
       "      <th>Transcript type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>894636</td>\n",
       "      <td>894637</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>ENSG00000188976.11</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>ENST00000327044.7</td>\n",
       "      <td>944203</td>\n",
       "      <td>959309</td>\n",
       "      <td>944203</td>\n",
       "      <td>959256</td>\n",
       "      <td>2757</td>\n",
       "      <td>tsl1 (assigned to previous version 6)</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>895964</td>\n",
       "      <td>895965</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000187961</td>\n",
       "      <td>ENSG00000187961.14</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>ENST00000338591.8</td>\n",
       "      <td>960584</td>\n",
       "      <td>965719</td>\n",
       "      <td>960584</td>\n",
       "      <td>965719</td>\n",
       "      <td>2567</td>\n",
       "      <td>tsl1 (assigned to previous version 7)</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>901877</td>\n",
       "      <td>901878</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000187583</td>\n",
       "      <td>ENSG00000187583.10</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>ENST00000379410.7</td>\n",
       "      <td>966497</td>\n",
       "      <td>975865</td>\n",
       "      <td>966497</td>\n",
       "      <td>975108</td>\n",
       "      <td>2404</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>901882</td>\n",
       "      <td>901883</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000187583</td>\n",
       "      <td>ENSG00000187583.10</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>ENST00000379407.7</td>\n",
       "      <td>966497</td>\n",
       "      <td>975865</td>\n",
       "      <td>966502</td>\n",
       "      <td>975008</td>\n",
       "      <td>2194</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>5</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>935491</td>\n",
       "      <td>935492</td>\n",
       "      <td>4</td>\n",
       "      <td>-1</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>ENSG00000188290.10</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>ENST00000304952.10</td>\n",
       "      <td>998962</td>\n",
       "      <td>1000172</td>\n",
       "      <td>998964</td>\n",
       "      <td>1000111</td>\n",
       "      <td>899</td>\n",
       "      <td>tsl1</td>\n",
       "      <td>HES4</td>\n",
       "      <td>4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  tss_start  tss_end  uid  strand   Gene stable ID  \\\n",
       "0  chr1     894636   894637    0      -1  ENSG00000188976   \n",
       "1  chr1     895964   895965    1       1  ENSG00000187961   \n",
       "2  chr1     901877   901878    2       1  ENSG00000187583   \n",
       "3  chr1     901882   901883    3       1  ENSG00000187583   \n",
       "4  chr1     935491   935492    4      -1  ENSG00000188290   \n",
       "\n",
       "  Gene stable ID version Transcript stable ID Transcript stable ID version  \\\n",
       "0     ENSG00000188976.11      ENST00000327044            ENST00000327044.7   \n",
       "1     ENSG00000187961.14      ENST00000338591            ENST00000338591.8   \n",
       "2     ENSG00000187583.10      ENST00000379410            ENST00000379410.7   \n",
       "3     ENSG00000187583.10      ENST00000379407            ENST00000379407.7   \n",
       "4     ENSG00000188290.10      ENST00000304952           ENST00000304952.10   \n",
       "\n",
       "   Gene start (bp)  Gene end (bp)  Transcript start (bp)  Transcript end (bp)  \\\n",
       "0           944203         959309                 944203               959256   \n",
       "1           960584         965719                 960584               965719   \n",
       "2           966497         975865                 966497               975108   \n",
       "3           966497         975865                 966502               975008   \n",
       "4           998962        1000172                 998964              1000111   \n",
       "\n",
       "   Transcript length (including UTRs and CDS)  \\\n",
       "0                                        2757   \n",
       "1                                        2567   \n",
       "2                                        2404   \n",
       "3                                        2194   \n",
       "4                                         899   \n",
       "\n",
       "          Transcript support level (TSL) Gene name  Transcript count  \\\n",
       "0  tsl1 (assigned to previous version 6)     NOC2L                 6   \n",
       "1  tsl1 (assigned to previous version 7)    KLHL17                 5   \n",
       "2                                   tsl1   PLEKHN1                 5   \n",
       "3                                   tsl1   PLEKHN1                 5   \n",
       "4                                   tsl1      HES4                 4   \n",
       "\n",
       "  Transcript type  \n",
       "0  protein_coding  \n",
       "1  protein_coding  \n",
       "2  protein_coding  \n",
       "3  protein_coding  \n",
       "4  protein_coding  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg19_genes = pd.read_csv(src.interim_data_path / \"hg19_ensembl_gene_tss.bed\", sep='\\t', header=None, names = ['chr', 'tss_start', 'tss_end', 'uid'])\n",
    "hg19_genes = hg19_genes.merge(genes.drop(['chr', 'tss_start', 'tss_end'], axis=1), on='uid')\n",
    "hg19_genes.to_csv(src.interim_data_path / 'hg19_ensembl_gene_tss.tsv', sep='\\t', index=False, header=True)\n",
    "hg19_genes.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (hic_reconcile_ctcf)",
   "language": "python",
   "name": "hic_reconcile_ctcf"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
